composeApp/src/commonMain/kotlin/sp/bvantur/tasky/register/ui/RegisterScreen.kt

```diff
@@ -37,12 +46,22 @@ import tasky.composeapp.generated.resources.password
 
 @Composable
 fun RegisterRoute() {
-    RegisterScreen()
+    val viewModel = koinViewModel<RegisterViewModel>()
+    val viewState: RegisterViewState by viewModel.viewStateFlow.collectAsStateWithLifecycle()
+
+    RegisterScreen(
+        viewState = viewState,
+        onUserAction = viewModel::onUserAction
+    )
 }
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
-fun RegisterScreen() {
+fun RegisterScreen(viewState: RegisterViewState, onUserAction: OnRegisterUserAction) {
+    val name = rememberSaveable { mutableStateOf("") }
+    val email = rememberSaveable { mutableStateOf("") }
+    val password = rememberSaveable { mutableStateOf("") }
```

Comment:
If these `rememberSaveable` states are just for temporary testing, please ignore this comment :D 

I strongly recommend putting all UI states in a ViewModel. With `rememberSaveable,` your state will survive configuration changes, but then your UI will have too many responsibilities since it also needs to decide how the state is updated.